File: AdventJS/2025/reto 7/7.py
def draw_tree(height, ornament, frequency):
  tree = []
  c = 1
  for i, k in enumerate(range(height, 0, -1)):
    tree.append(" " * (k - 1))
    for j in range(2*i + 1):
      tree.append(ornament if c % frequency == 0 else "*")
		  c += 1
    tree.append("\n")
  
  tree.pop()
  tree.append("\n" + " " * (height - 1) + "#")
  return "".join(tree)

File: AdventJS/2025/reto 7/7.js
/** @param {number} height - Height of the tree
 *  @param {string} ornament - Character to use as ornament
 *  @param {number} frequency - How often ornaments appear
 *  @returns {string} The decorated tree
 */
function drawTree(height, ornament, frequency) {
  let branch = "";
  let c = 1;

  for (let i = 0, k = height; k > 0 && i < height; i++, k--) {
    branch += " ".repeat(k - 1);
    for (let j = 0; j < (2*i + 1); j++) {
      branch += (c % frequency === 0) ? ornament : "*";
      c++;
    }
    branch += "\n";
  }

  branch += " ".repeat(height - 1) + "#";
  return branch
}